src/app/webserver/app-pools/app-pools.service.ts

```diff
@@ -123,26 +123,17 @@ export class AppPoolsService {
         });
     }
 
-    recycle(pool: ApplicationPool) {
+    recycle(pool: ApplicationPool): Promise<any> {
         if (pool.status == Status.Stopped) {
             return this.start(pool);
+        } else {
+            return this.restart(pool);
         }
+    }
 
-        return new Promise<ApplicationPool>((resolve, reject) => {
-            this.stop(pool).then(_ => {
-                this.start(pool).then(
-                    p => resolve(p),
-                ).catch(e => reject(e));
-            }).catch(e => reject(e));
-            if (pool.status == Status.Stopping) {
-                let ob = interval(200).subscribe(i => {
-                    if (pool.status == Status.Stopped || i >= 450) {
-                        ob.unsubscribe();
-                        this.start(pool);
-                    }
-                });
-            }
-        });
+    async restart(pool: ApplicationPool): Promise<any> {
```

Comment:
In the previous code, we handles exception error and do some retry logic. But in the new code, there is no such a behavior.